### ISSUE TITLE
[Feature Request] Global Effect: Output Ceiling

### ISSUE THREAD

**Issue by Phazmaonline** [jerryuhoo/Fire#37]
While playing with multiband distortion I have noticed that it is easy to get a pretty high peak level on the output. I think it would be nice if we had an option to link the global output level to some sort of safety limiter or clipper ceiling, meaning we will never get higher levels than what we set as Output dB.

**Comment by jerryuhoo** [comment]
Like adding a limiter in case of the level is over 0db? That sounds doable.

**Comment by Phazmaonline** [comment]
Yes something like that. I would put it as last step in the signal chain before the global output level knob. (So that when i set the output level to -6dB for example, I know that the audio coming out of the plugin will never be higher than -6dB).

**Comment by jerryuhoo** [comment]
I just released [v1.0.0](https://github.com/jerryuhoo/Fire/releases/tag/v1.0.0), feel free to reopen this issue if you have any suggestions or questions.

**Comment by Phazmaonline** [comment]
Thanks for adding the limiter. Something seems to not work as expected though. There is always an initial transient that slips through and goes up to 0dB, even if the threshold is lower. And after that initial part the signal is limited to a level which is lower than the chosen threshold. I recorded a video that shows it - as you can see with this particular sound the output is always 2.5dB lower than the threshold.

https://user-images.githubusercontent.com/93682581/181813773-886ed514-f0fd-4c5c-a00e-a265fe16a41b.mov

**Comment by jerryuhoo** [comment]
Dang! I used the juce::dsp::limiter and it seems not the original limiter that I thought. It has auto make-up so it is louder than normal. Do you think it is good to use a compressor in the global panel and you can change the ratio to be either a compressor or a limiter (when the ratio is larger than 20:1)?
